### DISCUSSION TITLE
[Windows] MainPage.Background does not react on Theme change

### DISCUSSION THREAD

**Discussion by kucint** [unoplatform/uno#17487] [Q&A] [answered]
Changing Theme to dark and back to light works fine on `Desktop`.
On `Windows`, controls seem to react correctly on change, except the main frame background which stays always light.

```cs
    public MainPage() => this
        .DataContext(new BindableMainPageModel(), (page, vm) => page
        .Background(ThemeResource.Get<Brush>("ApplicationPageBackgroundThemeBrush"))   // -> does not work on windows
        .Content(
            new ToggleSwitch()
            .OnContent("Dark mode")
            .OffContent("Light mode")
            .Name(out ToggleSwitch c, c => c.Toggled += (s, e) =>
            {
                FrameworkElement? content = XamlRoot?.Content as FrameworkElement;
                if (content is not null)
                    content.RequestedTheme = content.RequestedTheme == ElementTheme.Dark ? ElementTheme.Light : ElementTheme.Dark;
            })
            .IsOn(x => x.Binding(() => vm.IsAppInDarkMode).TwoWay())
        ));
```

Model:
```cs
internal partial record MainPageModel
{
    public IState<bool> IsAppInDarkMode => State<bool>.Value(this, () => false);
}
```
![image](https://github.com/user-attachments/assets/ac1640be-c19d-4a5c-ae41-198cd898ee59)

![image](https://github.com/user-attachments/assets/587a6717-0780-4cf4-b916-dc79c9d9f31e)

> [!NOTE]
> Latest dev branch used: `"Uno.Sdk": "5.4.0-dev.99"`

**Comment by morning4coffe-dev** [comment, marked as answer]
Hey @kucint, I managed to reproduce the issue using MVUX+CSM, and it appears there's a problem with how the resources are currently applied. If you notice, on Desktop (and other targets) although the theme updates, the Dark theme seems to be black instead of the dark gray used with Fluent. 

However, I couldn't replicate the same issue with MVVM+Xaml. I've created an issue in uno.extensions repo to track this further: https://github.com/unoplatform/uno.extensions/issues/2429
